terra_layer/style.py

```diff
@@ -482,290 +509,579 @@ def gen_legend_circle(
+        return {
+            "items": gen_color_legend_steps(
+                boundaries,
+                colors,
+                no_value,
+            )[::-1],
+        }
+    else:
+        color = colors[0]
+        if no_value:
+            color = no_value
+
+        return {
+            "items": [
+                {
+                    "color": color,
+                    "boundaries": {
+                        "lower": {"value": None, "included": True},
+                        "upper": {"value": None, "included": True},
+                    },
+                    "shape": style_type_2_legend_shape.get(map_style_type, "square"),
                 }
-            else:
-                config_style_no_value = DEFAULT_STYLE_NO_VALUE
-
-            field_style = get_field_style(field)
-            style = gen_layer_fill(
-                field_style,
-                fill_color=gen_style_steps(field_style, boundaries, colors),
-                style=config_style,
-                include_no_value=include_no_value,
-                style_no_value=config_style_no_value,
-            )
+            ]
+        }
 
-            legend_addition = {
-                "items": gen_legend_steps(
-                    boundaries,
-                    colors,
-                    include_no_value,
-                    config_style_no_value["fill_color"],
-                )[::-1],
-            }
-            return (style, legend_addition)
-        else:
-            return (
-                DEFAULT_STYLE_GRADUADED,
-                {"items": [DEFAULT_LEGEND_GRADUADED]},
-            )
 
-    elif symbology == "circle":
-        """config = {
-            "field": "my_field",
-            "symbology": "circle",
-            "include_no_value": False,  # Show no value features on map and legend
-            "max_diameter": 200,
-            "style": {
-                "fill_color": "#0000cc",
-                "fill_opacity": 0.5,
-                "stroke_color": "#ffffff",
-                "stroke_width": 1,
-            },
-            "no_value_style": {
-                "fill_color": "#000000",
-                "fill_opacity": 0,
-                "stroke_color": "#ffffff",
-                "stroke_width": 1,
-            },
+def gen_categorized_value_style(geo_layer, data_field, prop_config, default_no_value):
+    default_value = None
+    field_getter = ["get", data_field]
+
+    if not prop_config["categories"]:
+        return None
+
+    steps = ["case"]
+    for category in prop_config["categories"]:
+        name = category["name"]
+        value = category["value"]
+        if name is None:
+            default_value = value
+            continue
+
+        steps.append(["==", field_getter, name])
+        steps.append(value)
+
+    steps.append(default_value or default_no_value)
+
+    if default_value is not None:
+        return ["case", ["has", data_field], steps, default_value]
+    else:
+        return steps
```

Comment:
Use `match` in place of `case`.
https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#match

Reply:
right !

Reply:
Done